homeassistant/components/webostv/config_flow.py

```diff
@@ -0,0 +1,185 @@
+"""Config flow to configure webostv component."""
+from urllib.parse import urlparse
+
+from aiopylgtv import PyLGTVPairException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import ssdp
+from homeassistant.const import CONF_CLIENT_SECRET, CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from . import CannotConnect, async_control_connect
```

Comment:
Since the library already throws known exceptions you could avoid creating another one which may limit the ability to debug what was going on since it will swallow the other ones.

```
diff --git a/homeassistant/components/webostv/__init__.py b/homeassistant/components/webostv/__init__.py
index 02b12b5d01..1759e9a940 100644
--- a/homeassistant/components/webostv/__init__.py
+++ b/homeassistant/components/webostv/__init__.py
@@ -85,10 +85,6 @@ SERVICE_TO_METHOD = {
 _LOGGER = logging.getLogger(__name__)
 
 
-class CannotConnect(Exception):
-    """Error to indicate we cannot connect."""
-
-
 async def async_setup(hass, config):
     """Set up the LG WebOS TV platform."""
     hass.data.setdefault(DOMAIN, {})
@@ -193,8 +189,6 @@ async def async_control_connect(host: str, key: str) -> WebOsClient:
     except PyLGTVPairException as err:
         _LOGGER.warning("Connected to LG webOS TV %s but not paired", host)
         raise PyLGTVPairException(err) from err
-    except WEBOSTV_EXCEPTIONS as err:
-        raise CannotConnect(f"Error connecting to LG webOS TV {host}") from err
 
     return client
 
diff --git a/homeassistant/components/webostv/config_flow.py b/homeassistant/components/webostv/config_flow.py
index 32cea95312..2683d0330f 100644
--- a/homeassistant/components/webostv/config_flow.py
+++ b/homeassistant/components/webostv/config_flow.py
@@ -10,7 +10,7 @@ from homeassistant.const import CONF_CLIENT_SECRET, CONF_HOST, CONF_NAME
 from homeassistant.core import callback
 from homeassistant.helpers import config_validation as cv
 
-from . import CannotConnect, async_control_connect
+from . import async_control_connect
 from .const import CONF_ON_ACTION, CONF_SOURCES, DEFAULT_NAME, DOMAIN, ON_ACTION_DOCS
 
 DATA_SCHEMA = vol.Schema(
@@ -68,7 +68,7 @@ class FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
                 client = await async_control_connect(self._user_input[CONF_HOST], None)
             except PyLGTVPairException:
                 return self.async_abort(reason="error_pairing")
-            except CannotConnect:
+            except WEBOSTV_EXCEPTIONS:
                 errors["base"] = "cannot_connect"
             else:
                 return await self.async_step_register(self._user_input, client)
@@ -174,7 +174,7 @@ async def async_default_sources(host, key) -> list:
     sources = []
     try:
         client = await async_control_connect(host, key)
-    except CannotConnect:
+    except WEBOSTV_EXCEPTIONS:
         return None
 
     for app in client.apps.values():
~

```

Reply:
Indeed it makes it aligned with other places, fixed